nio.conf

```diff
@@ -1,103 +1,276 @@
 [DEFAULT]
 
-[provider]
-conf: etc
-type: niocore.configuration.providers.file.FileConfigurationProvider
+[environment]
+
+# Location of custom extensions for auto-discovery
+#
+#blocks=blocks
+#components=niocore.components
+#services=nio.service
+
+[components]
 
-[modules]
-# location of the module configuration files
-conf: etc/modules
-load: persistence, scheduler, security, web, communication
-# comma separated list of module locations
-locations: niocore.modules, nio.modules
+# discoverable components that should be disabled from project
+#
+#disabled=SNMPAgent
 
 [service]
-# location of overall service configuration
-conf: etc/service.cfg
-block_router: nio.router.thread_pool_executor.ThreadedPoolExecutorRouter
-# block_router: nio.router.threaded.ThreadedBaseBlockRouter
-async_start: true
+
+# specify block router class used by services
+#
+#block_router=nio.router.thread_pool_executor.ThreadedPoolExecutorRouter
+
+# when launching nio, start auto-start services asynchronously
+#
+#async_start=true
+
+# clone signals when a block notifies to two blocks
+#
+#clone_signals=true
+
+# block router verifies that objects processed by blocks are nio signals
+#
+#check_signal_type=true
+
+# when using ThreadedPoolExecutorRouter max workers is the maximum number of
+# threads that will run concurrently
+#
+#thread_pool_max_workers=50
+
+# time to wait for a service's response when sending an IPC request
+#
+#ipc_request_time_to_live=2
 
 [logging]
-conf: etc/logging.cfg
 
-[security]
+# dictionary of python logging configuration
+#
+conf=etc/logging.json
 
-# basic authentication
-users : etc/users.dat
+[providers]
 
-# oauth2 authentication
-# client_id = 407408718192.apps.googleusercontent.com
-# validate_uri = https://www.googleapis.com/oauth2/v2/tokeninfo?access_token=%s
+# file provider python module
+#
+#type=niocore.configuration.providers.file.FileConfigurationProvider
 
-# authorization
-permissions : etc/permissions.dat
+# path to configuration file provider
+#
+#config_directory=etc
 
 [communication]
-bind_ip_address: [[NIOHOST]]
-ip_address: [[NIOHOST]]
-xpub_port: [[XPUB_PORT]]
-xsub_port: [[XSUB_PORT]]
-matching: nio.modules.communication.matching.default.DefaultMatching
-# master: true
-# poll_rate_ms: 100
-# min_port: 9010
-# max_port: 9199
-# connect_timeout: 0
-# connect_check_interval: 1
-# network_monitor_timeout: 30
-# heartbeat_interval: 100
-# heartbeat_check_interval: 150
-# publisher_heartbeat_interval: 0
-# publisher_heartbeat_check_interval: 0
-# clients_gone_max_history: 0
-# clients_gone_sync_interval: 0
-# stats: false
+
+# is this the master broker or a slave node
+#
+#master=True
+
+# the ip address of the master broker
+#
+broker_ip_address=[[COMHOST]]
+
+# your public ip address that the master broker can talk to you on
+#
+ip_address=0.0.0.0
```

Comment:
Just verified with a master and slave and it works fine keeping the setting at 0.0.0.0 (both instances running on the same computer)